LAKMobile/src/screens/EditProfileScreen.tsx

```diff
@@ -0,0 +1,385 @@
+            if (response == null) {
+                return;
+            }
+            setProfileData(updatedUser);
+            navigation.navigate("ProfileScreen");
+            
+        })
+    }
+    
+    const isUserTheViewer = getCurrentUser() === route.params.userId;
+
+    return (
+        <View style={styles.mainContainer}>
+            <ScreenHeader showArrow={true} children={"Edit Profile"}/>
+            <ScrollView style={styles.scrollContainer}>
+                <View style={styles.nameContainer}>
+                    <PublicProfilePicDefault/>
+                </View>
+
+                <LabelWrapper label='User Name'>
+                    <TextInput
+                    style={styles.input}
+                    keyboardType="default"
+                    defaultValue={profileData?.firstName + " " + profileData?.lastName}
+                    onChangeText={(value) => setUserName(value)}
+                    />
+                </LabelWrapper>
+                
+                <View style={[styles.sectionTitleContainer]}>
+                    <AppText style={styles.sectionTitle}>Contact Information</AppText>
+                </View>
+
+                <LabelWrapper label='Mobile Number'>
+                    <TextInput
+                    style={bigInputStyle}
+                    keyboardType="default"
+                    defaultValue={profileData?.phone}
+                    onChangeText={(value) => setPhoneNumber(value)}
+                    />
+                </LabelWrapper>
+
+                <LabelWrapper label='Location'>
+                    <TextInput
+                    style={bigInputStyle}
+                    keyboardType="default"
+                    defaultValue={profileData?.location}
+                    value={location}
+                    onChangeText={(value) => setLocation(value)}
+                    />
+                </LabelWrapper>
+                <View style={[styles.pickerWrapper, styles.spacer]}>
+                    <Picker
+                    mode="dropdown" // Android only
+                    onValueChange={(itemValue : string) => setDistrict(itemValue)}
+                    selectedValue={district}
+                    >
+                    {LOCATIONS.map((location, index) => (
+                        <Picker.Item key={index} label={location} value={location}/>
+                    ))}
+                    </Picker>
+                </View>
+                {profileData?.vehicleData && 
+                    <View>
+                        <View style={[styles.sectionTitleContainer]}>
+                            <AppText style={styles.sectionTitle}>Vehicle Information</AppText>
+                        </View>
+
+                        <LabelWrapper label='Type'>
+                            <View style={[styles.pickerWrapper, styles.spacer, {width: '45%'}]}>
+                                <Picker
+                                mode="dropdown" // Android only
```

Comment:
Just curious - what does Android only mean here? Will this work on other devices?

Reply:
Android only means that this is a property that will only be read by an Android device. For all other devices, it should run as intended and simply ignore that line.